src/main/java/uk/gov/companieshouse/api/accounts/validation/AccountsValidator.java

```diff
@@ -1,27 +1,65 @@
 package uk.gov.companieshouse.api.accounts.validation;
 
+import javax.servlet.http.HttpServletRequest;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 import uk.gov.companieshouse.api.accounts.exception.DataException;
+import uk.gov.companieshouse.api.accounts.links.CompanyAccountLinkType;
+import uk.gov.companieshouse.api.accounts.model.rest.CompanyAccount;
+import uk.gov.companieshouse.api.accounts.model.rest.SmallFull;
 import uk.gov.companieshouse.api.accounts.model.validation.Errors;
+import uk.gov.companieshouse.api.accounts.service.CompanyAccountService;
+import uk.gov.companieshouse.api.accounts.service.CompanyService;
+import uk.gov.companieshouse.api.accounts.service.impl.SmallFullService;
+import uk.gov.companieshouse.api.accounts.service.response.ResponseObject;
 import uk.gov.companieshouse.api.accounts.validation.transactionclosure.CurrentPeriodTnClosureValidator;
+import uk.gov.companieshouse.api.accounts.validation.transactionclosure.PreviousPeriodTnClosureValidator;
 import uk.gov.companieshouse.api.model.transaction.Transaction;
 
-import javax.servlet.http.HttpServletRequest;
-
 @Component
-public class AccountsValidator {
+public class AccountsValidator extends BaseValidator {
+
+    private static final String SMALL_FULL_PATH = "$company_accounts.small_full";
+
+    private CompanyAccountService companyAccountService;
+
+    private SmallFullService smallFullService;
 
-    @Autowired
     private CurrentPeriodTnClosureValidator currentPeriodTnClosureValidator;
 
+    private PreviousPeriodTnClosureValidator previousPeriodTnClosureValidator;
+
+    @Autowired
+    public AccountsValidator(CompanyService companyService, CompanyAccountService companyAccountService,
+            SmallFullService smallFullService, CurrentPeriodTnClosureValidator currentPeriodTnClosureValidator,
+            PreviousPeriodTnClosureValidator previousPeriodTnClosureValidator) {
+        super(companyService);
+        this.companyAccountService = companyAccountService;
+        this.smallFullService = smallFullService;
+        this.currentPeriodTnClosureValidator = currentPeriodTnClosureValidator;
+        this.previousPeriodTnClosureValidator = previousPeriodTnClosureValidator;
+    }
+    
     public Errors validationSubmission(Transaction transaction, String companyAccountsId, HttpServletRequest request)
```

Comment:
validationSubmission isn't a verb (i.e. normal method name). Is this just local convention?

Reply:
... and yes, I know that this is preexisting code.